core/data_restore/src/storage_interactor.rs

```diff
@@ -27,15 +27,16 @@ pub fn update_tree_state(
     block_number: u32,
     account_updates: &[(u32, AccountUpdate)],
     connection_pool: ConnectionPool,
-) -> Result<(), DataRestoreError> {
-    let storage = connection_pool.access_storage().map_err(|_| {
-        DataRestoreError::Storage(
-            "db connection failed for data restore save tree state".to_string(),
+) -> Result<(), failure::Error> {
+    let storage = connection_pool.access_storage().map_err(|e| {
+        format_err!(
+            "Db connection failed for data restore save tree state: {}",
+            e.to_string()
         )
     })?;
     storage
         .update_tree_state(block_number, &account_updates)
-        .map_err(|_| DataRestoreError::Storage("cant save tree state".to_string()))?;
```

Comment:
there was no need to remove `DataRestoreError` - it was not bad at all.
you could use `#[derive(Fail)]` see example https://github.com/rust-lang-nursery/failure

Reply:
Agree, but I have not seen other error enums anywhere else in the code. And now I see it as an "extra code". Think we can use failure::Error everywhere